### PR TITLE
Revisit Functional Test Pattern Part 1: The Refactoring

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Add SharedStoreFixtureBase
 Derive F1FixtureBase from SharedStoreFixtureBase
 Make TestModelSource use ModelCustomizer

 Part of #9098

### CODE REVIEW

File: src/EFCore/Infrastructure/ModelSource.cs [hunk 1]
```diff
@@ -73,17 +73,16 @@ public virtual IModel GetModel(DbContext context, IConventionSetBuilder conventi
...
-            var internalModelBuilder = ((IInfrastructure<InternalModelBuilder>)modelBuilder).Instance;
-
-            internalModelBuilder.Metadata.SetProductVersion(ProductInfo.GetVersion());
+            var model = modelBuilder.GetInfrastructure().Metadata;
+            model.SetProductVersion(ProductInfo.GetVersion());
 
             Dependencies.ModelCustomizer.Customize(modelBuilder, context);
 
-            internalModelBuilder.Validate();
+            model.Validate();
```
Comment (smitpatel) [inline src/EFCore/Infrastructure/ModelSource.cs]: Remove `IMB.Validate`

**Comment by smitpatel** [comment]
:shipit:

File: src/EFCore.Relational.Specification.Tests/IRelationalTestStore.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+namespace Microsoft.EntityFrameworkCore
+{
+    public interface IRelationalTestStore<out TConnection>
+        where TConnection : DbConnection
+    {
+        TConnection Connection { get; }
+        string ConnectionString { get; }
+        DbTransaction Transaction { get; }
+
+        void Dispose();
```
Comment (ajcvickers) [inline src/EFCore.Relational.Specification.Tests/IRelationalTestStore.cs]: Should this interface be IDisposable?

File: src/EFCore.Relational.Specification.Tests/Query/NullSemanticsQueryTestBase.cs [hunk 3]
```diff
@@ -653,7 +653,7 @@ public virtual void From_sql_composed_with_relational_null_comparison()
             using (var context = CreateContext(useRelationalNulls: true))
             {
                 var actual = context.Entities1
-                    .FromSql(@"SELECT * FROM ""NullSemanticsEntity1""")
+                    .FromSql(@"SELECT * FROM ""Entities1""")
```
Comment (ajcvickers) [inline src/EFCore.Relational.Specification.Tests/Query/NullSemanticsQueryTestBase.cs]: What causes this change?

Comment (ajcvickers) [inline src/EFCore.Relational.Specification.Tests/Query/NullSemanticsQueryTestBase.cs]: Never mind, figured it out.

Comment (AndriySvyryd) [inline src/EFCore.Relational.Specification.Tests/Query/NullSemanticsQueryTestBase.cs]: Using ModelCustomizer, it configures table names as DbSet names

**Pull request merged**